eng/testing/tests.mobile.targets

```diff
@@ -131,7 +141,8 @@
       <_runnerFilesToPublish Include="$(AppleTestRunnerDir)*" Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS'" />
       <_runnerFilesToPublish Include="$(WasmTestRunnerDir)*" Condition="'$(TargetOS)' == 'Browser'" />
 
-      <ResolvedFileToPublish Include="@(_runnerFilesToPublish)" RelativePath="%(FileName)%(Extension)" CopyToPublishDirectory="PreserveNewest" PostprocessAssembly="true" />
+      <!-- Exclude xunit assemblies as those should be resolved by our own package references -->
+      <ResolvedFileToPublish Include="@(_runnerFilesToPublish)" Condition="!$([System.String]::Copy('%(Filename)').StartsWith('xunit.'))" RelativePath="%(FileName)%(Extension)" CopyToPublishDirectory="PreserveNewest" PostprocessAssembly="true" />
```

Comment:
Don't understand the ocmment. What is our own package reference?

Reply:
We get this from the test runners package reference (Microsoft.DotNet.XunitRunners or something like that), right?

Reply:
which files do you want to exclude here? the xunit.console.dll?

Reply:
Yeah... because those are also package references to the XharnessRunners which is referenced by our mobile test runners:

![image](https://user-images.githubusercontent.com/22899328/84811105-3884d600-afc1-11ea-818f-9c3d8065f372.png)

This is effectively fixing a Double Write @akoeplinger saw in the binlog.

Reply:
The xharness nugets specify nuget package references for the xunit packages, the issue is that we build the WasmTestRunner .csproj which references the xharness nuget and that means in the output folder we get all these assemblies.

Later on when we build the test assembly we reference the WasmTestRunner project, so all its output assemblies incl. xunit dlls get copied, but we also have them a second time from the test assembly PackageReference.

Not sure if there's a better way to solve that.

Reply:
Hard to say... If the test runners would be consumed via a package then NuGet would be able to resolve the transitive dependencies up-front. I still don't have all the details here, how would customers test their wasm/mobile application? Presumably with VSTest? Do we plan to integrate these runner into VSTest at some point?

Reply:
> If the test runners would be consumed via a package then NuGet would be able to resolve the transitive dependencies up-front.

They're built locally at the moment and we copy them to the publish dir because we wanted to avoid adding P2P to all test projects, so that way NuGet would resolve the transitive dependencies and ambiguities.

> I still don't have all the details here, how would customers test their wasm/mobile application? Presumably with VSTest? Do we plan to integrate these runner into VSTest at some point?

That I'll defer to @akoeplinger

Reply:
> I still don't have all the details here, how would customers test their wasm/mobile application? Presumably with VSTest? Do we plan to integrate these runner into VSTest at some point?

Right now these runners are for our own (dotnet/runtime) usage to run our tests and only support the subset of functionality we need. Higher-level SDKs that build on our runtime packs like Xamarin or Blazor already have their own ways how to package apps.

VSTest support is probably relevant at some point in the future, not clear to me yet how that would work.